mpf/devices/autofire.py

```diff
@@ -47,6 +47,13 @@ def _initialize(self) -> None:
                 self.config['ball_search_order'], self._ball_search, self.name)
         # pulse is handled via rule but add a handler so that we take notice anyway
         self.config['switch'].add_handler(self._hit)
+        if self.config['enable_timeouts']:
+            self._timeout_watch_time = self.config['timeout_watch_time']/1000
+            self._timeout_max_hits = self.config['timeout_max_hits']
+            self._timeout_disable_time = self.config['timeout_disable_time']/1000
+            self._timeout_hits = []
+            from time import time as _time
+            from threading import Timer as _Timer
```

Comment:
This timer will interfere with asyncio. You can use self.machine.clock instead or instantiate DelayManager(self.machine.delayRegistry) for a nicer interface.